src/interpreter/plugin/BooleanPlugin.ts

```diff
@@ -142,7 +142,7 @@ export class BooleanPlugin extends FunctionPlugin implements FunctionPluginTypec
    */
   public and(ast: ProcedureAst, state: InterpreterState): InterpreterValue {
     return this.runFunction(ast.args, state, this.metadata('AND'),
-      (...args) => !args.some((arg: boolean) => !arg)
+      (...args) => args.filter((arg: boolean | undefined) => arg !== undefined).every((arg: boolean) => !!arg)
```

Comment:
TODO: test it

Reply:
done

Reply:
It's a bit more complicated. My refactor of `FunctionPlugin` changed the way strings are coerced to boolean values, which implicitly changed the behavior of boolean functions (e.g. `AND`). For this reason, this tiny modification in `BooleanPlugin` was necessary to keep the current behavior.